.circleci/config.yml

```diff
@@ -340,11 +417,89 @@ jobs:
         type: boolean
         default: false
     steps:
+      # Section 1: Prepare environment
       - checkout_with_submodules
       - python/install-packages:
           pkg-manager: pip
-      - when:
-          condition: << parameters.has_results >>
+      - run:
+          name: Setup Bash Environment
+          command: |
+            # set -x  # Uncomment to debug
+            TMP_ENV=$(mktemp)
+
+            if [ "<<# parameters.allow_fail >>allow_fail<</ parameters.allow_fail >>" == "allow_fail" ]
```

Comment:
(suggestion, non-blocking) Add comment on the [mustache conditionals](https://support.circleci.com/hc/en-us/articles/4417604103835-Using-Mustache-Conditionals-in-Config-File) and/or consider using `steps` and `condition`.

Reply:
The main documentation mentions conditions, and the "mustache" syntax is barely mentioned. However, it is a big change to switch to conditions, and it will take some experimentation to make it look decent. I think I'll do the small thing, adding some comments, and tackle the re-write in a new PR, so we can decide if it is an improvement.